### PR TITLE
Install Stave via pip

### LINKED ISSUE #183: Install Stave via pip
**Is your feature request related to a problem? Please describe.**
Currently users need to manually install `django`, `react.js` and other dependencies
 to boot a Stave instance. Is there any way we can wrap it to a standalone package that can be `pip` installed.

**Describe the solution you'd like**
Package the Stave project with required dependencies according to [this tutorial](https://packaging.python.org/tutorials/packaging-projects/).

**Describe alternatives you've considered**
May need to include the `build` folder for `react.js` rendering and `db.sqlite3` for `django` backend.

**Additional context**
Should support viewer mode and full mode.

### PR DESCRIPTION
**Pull request by mylibrar**
This PR fixes #183. 

### Description of changes
- Add **`setup.py`** and **`MANIFEST.in`** to package Stave. `MANIFEST.in` includes two package data:
    - Copy **`db.sqlite3`** to `stave/simple-backend/nlpviewer_backend` and modify **`settings.py`** to reroute database to the new path.
    - Copy **`build`** folder to `stave/simple-backend/nlpviewer_backend/lib/`.
- Update `StaveViewer` interface: remove the `build_path` parameter since it is already copied to the same directory of **`stave_viewer.py`**. **`manageproject.py`** is also updated to reflect this change.

### Possible influences of this PR
Larger repo size because of the `build` folder. May be harder to maintain since every update to the frontend code should be manually migrated to the `build` folder under `stave/simple-backend/nlpviewer_backend/lib/` (by running `npm run build` and then copy the `build/` to the package path).

### Demonstration of results
- Install
    - Approach 1: Go to `stave/` and run `pip3 install .`.
    - Approach 2: The source archive and built distribution are already uploaded to [TestPyPI](https://test.pypi.org/project/stave/). Run the following command to install:
        > pip3 install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple stave 

- Test Snippet
    > from nlpviewer_backend.lib.stave_viewer import StaveViewer
sv = StaveViewer(in_viewer_mode=False)
sv.run()
sv.open()

### CODE REVIEW

**Review by hunterhector — COMMENTED** [review]
There are a few things that I am not sure whether we want to include in the PR:

1. The database. We previously did not use git control the binary database. You can also see that the size of the database is 25 MB, which will increase the burden for package installing. We instead use the SQL queries to create the database on the fly: https://github.com/asyml/stave/blob/master/simple-backend/start-dev.sh#L5. Can we do the same here? The hope is to keep the installation lightweight, and then do not ask Git to track binary files.
2. The build directory. I presume the build allows the user to use Stave without installing stuff like yarn, node, react. But it seems that they are not suitable for version control. Is it possible that we place the binary files into another downloadable path? One idea I have is to put them as `assets` of github releases like here: https://github.com/asyml/stave/archive/refs/tags/v0.0.1-alpha.zip   Then we simply download the release during install?


By the way, if we would to make these changes, we probably would need to use a fresh branch. This branch has already been pushed with binary files and new commits won't overwrite them, which may cause an increase in Git history.

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    packages=setuptools.find_packages(where="simple-backend"),
+    include_package_data=True,
+    platforms='any',
+
+    install_requires=[
+        'django>=3.0.4',
+        'django-guardian==2.3.0',
+        'tornado==6.1'
+    ],
+    extras_require={},
```
Comment (hunterhector) [inline setup.py:31]: We can add `Forte` as extra requires.

**Comment by hunterhector** [comment]
> May be harder to maintain since every update to the frontend code should be manually migrated to the `build` folder under `stave/simple-backend/nlpviewer_backend/lib/` (by running `npm run build` and then copy the `build/` to the package path).

We can automate this process as a Github action like here: https://github.com/asyml/forte/blob/master/.github/workflows/main.yml#L150

So it can automatically build the project if you create a tag, and upload it to PyPI as a version. You can also make an individual workflow that can be triggered manually: https://github.com/asyml/forte/blob/master/.github/workflows/pypi.yml

File: scripts/settings.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+settings_dir = os.path.dirname(os.path.abspath(__file__))
+BASE_DIR = os.path.dirname(settings_dir)
+
+DATA_UPLOAD_MAX_MEMORY_SIZE = 5242880
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.1/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = '_uc(72pjf1%e_@%)j=pl%ag52_flmv++&e%frth__c%io9bom^'
```
Comment (hunterhector) [inline scripts/settings.py:25]: We need to encrypt this key, do we?

**Review by hunterhector — COMMENTED** [review]
I haven't tested out this new PR yet, but according to [here](https://stackoverflow.com/questions/9498201/delete-file-from-pull-request-on-github), this PR will still contain the files you removed in the history (which is about 25 MB). I think it is better to restart a new PR from a new branch, and you can copy the changes there.

**Comment by mylibrar** [comment]
This PR is migrated to #186.

**Pull request closed without merging**